pyiceberg/table/update/validate.py

```diff
@@ -69,3 +75,74 @@ def validation_history(
         raise ValidationException("No matching snapshot found.")
 
     return manifests_files, snapshots
+
+
+def deleted_data_files(
+    table: Table,
+    starting_snapshot: Snapshot,
+    data_filter: Optional[BooleanExpression],
+    parent_snapshot: Optional[Snapshot],
+    partition_set: Optional[set[Record]],
```

Comment:
This looks out of order:
```suggestion
    partition_set: Optional[set[Record]],
    parent_snapshot: Optional[Snapshot],
```

It looks like the function call in `validate_deleted_data_files` is assuming that `parent_snapshot` is the last argument
```
    conflicting_entries = deleted_data_files(table, starting_snapshot, data_filter, None, parent_snapshot)
```